src/DGtal/base/Alias.h

```diff
@@ -199,61 +199,67 @@ user forward an Alias<T> parameter.
     inline ~Alias() {}
 
     /**
-       Constructor from const reference to an instance of T. Invalid.
+       Constructor from const reference to an instance of T. Deleted.
+
+       Aliasing a const-ref is an error. Consider ConstAlias instead.
     */
-    inline Alias( const T& ) : myParam( CONST_LEFT_VALUE_REF ), myPtr( 0 )
-    { ASSERT(( false && "[Alias::Alias( const T& )] Aliasing a const-ref is an error. Consider ConstAlias instead." )); }
+    Alias( const T& ) = delete;
```

Comment:
We could just not declare those constructors in order to raise a compilation error when used (instead of an error at run-time in Debug mode) but declaring it with `= delete` makes sure that no other constructor will be chosen with const-ref and makes this choice more clear (and documented).

If the question is about the `delete` keyword, it is a new usage in `C++11` and it allows to explicitly say that a function or a member cannot be used (is deleted). For example, it replaces more efficiently the use of defined (but not implemented) private members (like when a class is not copyable).

The other new associated keyword is `default` to explicitly say you want the default behavior for constructor, destructor and assignment/move operators.